### DISCUSSION TITLE
I can't understand what the problem is.

### DISCUSSION THREAD

**Discussion by SomeOne-49** [vercel/next.js#50432] [Help]
### Summary

"For some reason, the value I pass through props in getStaticProps is not appearing in the component."

### Additional information

```js
This is my code :
import MeetupList from "@components/meetups/MeetupList";

const DUMME_MEETUPS = [
  {
    id: 1,
    title: "test1",
    image:
      "https://149541961.v2.pressablecdn.com/wp-content/uploads/2021/03/Meetup_logo.png",
    address: "123",
  },
  {
    id: 2,
    title: "test2",
    image:
      "https://149541961.v2.pressablecdn.com/wp-content/uploads/2021/03/Meetup_logo.png",
    address: "456",
  },
];

const AllMeetups = ({ meetups }) => {
  console.log(meetups);
  return <MeetupList meetups={meetups} />;
};


export async function getStaticProps () {
  return {
    props: {
      meetups: DUMME_MEETUPS
    }
  }
}
export default AllMeetups;

The Problem : 
Unhandled Runtime Error
Error: Cannot read properties of undefined (reading 'map')
   5 | return (
   6 |   <ul className={classes.list}>
>  7 |     {props.meetups.map((meetup) => (
     |                   ^
   8 |       <MeetupItem
   9 |         key={meetup.id}
  10 |         id={meetup.id}
```

**Comment by devjiwonchoi** [comment]
Since you have passed the `meetups` props to the `AllMeetups` component as an object destructuring assignment, 
the `props` which have been sent down to the `MeetupList` component are `meetups` itself. 

So when you try to map it inside the `MeetupList` component, it should be `props.map((meetup) => ...`

**Reply by SomeOne-49** [reply]
This is the output of `props `in the `MeetupList ` component: `{ meetups: undefined }
`

**Reply by devjiwonchoi** [reply]
First of all you should follow my instruction,
> So when you try to map it inside the `MeetupList` component, it should be `props.map((meetup) => ...`
then add a condition to check if props have been passed down to the component and is valid by doing this:

```
{ props ? props.map((meetup) => ( // ... )) : '' }
```

**Reply by devjiwonchoi** [reply]
The previous answer was just an easiest way to solve your problem and for more advanced way to solve it, 
and it is better to add a condition on the top of the function and check for the props have been passed down.

```
function MeetupList(props) {
if (!props) {
  // do something...
}

// ...

return (
    <ul className={classes.list}>
        {props.map((meetup) => (
// ...
```

**Comment by devjiwonchoi** [comment]
I assume that you are going to replace the `DUMME_MEETUPS` to `fetch data`, 
you will eventually going to use `try catch` while fetching the data. 
Therefore `returning the props "meetups"` should always be valid, and your current situation won't happen.
Or else it will be redirected to `404` page or something `fallback`ish.

**Comment by icyJoseph** [comment]
@SomeOne-49 Could you share your `_app` file? Most likely you are not passing `pageProps` along:

```jsx
export default function App({ Component, pageProps }) {
  return <Component {...pageProps} />;
}
```

In this context, `pageProps` is what either `getServerSideProps`, or `getStaticProps` return. So if you don't have it, it makes sense you won't see it at the page level.

Also make sure it is correctly spelled. `PageProps` won't work, it has to be `pageProps`.
